src/Symfony/Component/EventDispatcher/Attribute/AsEventListener.php

```diff
@@ -24,6 +24,10 @@ public function __construct(
         public ?string $method = null,
         public int $priority = 0,
         public ?string $dispatcher = null,
+        /** @var string|array{class?: class-string, service?: string, method?: string} */
```

Comment:
not `class-string|array{class?: class-string, service?: string, method?: string}` [as suggested](https://github.com/symfony/symfony/pull/50687#discussion_r1239646002) because we accept service id as well as class names